From 6420b46034e25f93cb25794c9ed37f03092d45d0 Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Fri, 28 Mar 2008 13:08:04 +0200
Subject: [PATCH] git-oodiff: layout fixes. Remove pythonism

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 git-oodiff |   90 ++++++++++++++++++++++++++++++++++--------------------------
 1 files changed, 51 insertions(+), 39 deletions(-)

diff --git a/git-oodiff b/git-oodiff
index 50ffe58..c19054b 100644
--- a/git-oodiff
+++ b/git-oodiff
@@ -5,55 +5,67 @@
 # 
 # $ GIT_EXTERNAL_DIFF=git-oodiff diff
 
+clean_up ()
+{
+    rm -f /tmp/oodiff.$$.{1,2}
+}
+
 convert_to_txt ()
 {
     if [ x"$1" = x"/dev/null" ]; then
-        printf "" > /tmp/oodiff.$$."$2"
+        : > /tmp/oodiff.$$."$2"
         eval "label$2=/dev/null/"
     else
         odt2txt "$1" > /tmp/oodiff.$$."$2" 2>/dev/null
     fi
 }
 
+Main ()
+{
+    if [ ! "$1" ]; then
+	echo "$0: [ERROR] Input argument missing" >&2
+	return 1
+    fi
+
+    label1="a/$1"
+    label2="b/$1"
 
-label1="a/$1"
-label2="b/$1"
-
-if [ "$#" = "1" ]; then
-    echo "Unmerged path $1"
-    exit 0
-fi
-
-msg=""
-if   [ x"$4" = x"." ]; then
-    msg="new file mode $7
-"
-elif [ x"$7" = x"." ]; then
-    msg="deleted file mode $4
-"
-elif [ x"$4" != x"$7" ]; then
-    msg="old mode $4\nnew mode $7
-"
-fi
-
-if convert_to_txt "$2" "1" &&
-    convert_to_txt "$5" "2" ; then
-    echo $(basename $0) "$label1" "$label2"
-    printf "%s" "$msg"
-    if diff -L "$label1" -L "$label2" -u /tmp/oodiff.$$.{1,2}; then
-        # no text change
-        if diff -q "$2" "$5"; then
-            : # no change at all
-        else
-            echo "OpenDocument files a/$1 and b/$1 files differ (same text content)"
-        fi
+    if [ "$#" = "1" ]; then
+	echo "Unmerged path $1"
+	exit 0
     fi
-else
-    # conversion failed. Fall back to plain diff.
-    echo diff -u "$label1" "$label2"
-    printf "%s" "$msg"
-    diff -L "$label1" -L "$label2" -u "$2" "$5"
-fi
 
-rm -f /tmp/oodiff.$$.{1,2}
+    msg=""
+
+    if [ x"$4" = x"." ]; then
+	msg="new file mode $7\n"
+    elif [ x"$7" = x"." ]; then
+	msg="deleted file mode $4\n"
+    elif [ x"$4" != x"$7" ]; then
+	msg="old mode $4\nnew mode $7\n"
+    fi
+
+    if convert_to_txt "$2" "1" &&
+       convert_to_txt "$5" "2"
+    then
+	echo $(basename $0) "$label1" "$label2"
+	echo -e "$msg"
+
+	if diff -L "$label1" -L "$label2" -u /tmp/oodiff.$$.{1,2}; then
+	    # no text change
+	    if diff -q "$2" "$5"; then
+		: # no change at all
+	    else
+		echo "OpenDocument files a/$1 and b/$1 files differ (same text content)"
+	    fi
+	fi
+    else
+	# conversion failed. Fall back to plain diff.
+	echo diff -u "$label1" "$label2"
+	echo -e "$msg"
+	diff -L "$label1" -L "$label2" -u "$2" "$5"
+    fi
+}
 
+Main "$@"
+trap 'clean_up; exit 0' 1 2 3 15
-- 
1.5.3.8

